ClipboardAPI/clipboard-change-event-explainer.md

```diff
@@ -0,0 +1,213 @@
+      - [Pros](#pros-2)
+      - [Cons](#cons-2)
+    - [4.2.2 Considered alternative: No focus requirement](#422-considered-alternative-no-focus-requirement)
+      - [Pros:](#pros)
+      - [Cons:](#cons)
+  - [4.3 Event bound to](#43-event-bound-to)
+  - [4.4 Event bubbles](#44-event-bubbles)
+  - [4.5 Event actions](#45-event-actions)
+  - [4.6 Event handler additional arguments](#46-event-handler-additional-arguments)
+- [5. Detailed design discussion](#5-detailed-design-discussion)
+  - [5.1 Listen to clipboard change directly from the OS](#51-listen-to-clipboard-change-directly-from-the-os)
+      - [Pros:](#pros-1)
+      - [Cons:](#cons-1)
+    - [5.1.1 APIs provided by all OS to listen to clipboardchange event:](#511-apis-provided-by-all-os-to-listen-to-clipboardchange-event)
+      - [Windows](#windows)
+      - [MacOS](#macos)
+      - [Linux X11/Wayland](#linux-x11wayland)
+      - [Android / iOS](#android--ios)
+      - [ChromeOS](#chromeos)
+  - [5.2 Considered alternative: Given the page focus restrictions, check clipboard hash change when page regains focus](#52-considered-alternative-given-the-page-focus-restrictions-check-clipboard-hash-change-when-page-regains-focus)
+      - [Pros:](#pros-2)
+      - [Cons:](#cons-2)
+- [6 References & acknowledgements](#6-references--acknowledgements)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## 1. Introduction
+
+The clipboardchange event fires whenever the system clipboard contents are changed. This allows web-apps like remote desktop clients to stay in sync with the system clipboard. It provides an efficient alternative to polling the clipboard for changes.
+
+
+## 2. Motivating Use Cases and scenarios
+
+### 2.1 Scenario: Sync clipboard with a remote desktop
+When a user copies text or an image on their local machine, the web-based remote desktop application can detect this clipboard change event through the browser's Clipboard API.
+Upon detecting the change, the application can automatically send the new clipboard content to the remote desktop environment.
+
+### 2.2 Scenario: Show available paste formats in web based editors
+Web based editors like Excel Online, Word Online may support paste operation in multiple formats. Within the UI, it may show the available formats on the UI (like csv, image, plain text). The clipboard change event can be used to detect the change in available formats in clipboard and reflect the same on the UI as soon as it is changed. 
+
+Similarly UI elements which depend on clipboard state, like "Paste image from clipboard" in an web based image editor, can be enabled/disabled using the clipboardchange event based on weather correct data format is present in clipboard or not.
+
+### 2.3 Alternative to inefficient polling of clipboard
+Today, a web-app can still monitor the system clipboard by polling and reading the clipboard through async clipboard API at regular intervals. 
```

Comment:
Proposing the following, can you please evaluate: 

"Today, a web-app can monitor the system clipboard by polling and reading the clipboard through async clipboard API at regular intervals. 
 However, polling is not efficient and this feature aims to introduce an efficient way notifying web apps when clipboard changes.
 Additionally we must ensure we monitor the clipboard only when absolutely required, that is, there is at least one document having required permissions and is listening to the clipboard change event. This will be described in design details."

Reply:
Sounds good, updated the content with very minor changes.